### PR TITLE
chore: warning when onboarding ids do not respect specific rules

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Checks if the ids of the steps of onboardings (sent to Telemetry) respect some specific rules, to make them more easily readable. In case of non respect, warnings are displayed in the backend console.

### What issues does this PR fix or reference?

Partially #4911

### CODE REVIEW

File: packages/main/src/plugin/onboarding-registry.ts [hunk 1]
```diff
@@ -142,4 +143,34 @@ export class OnboardingRegistry {
...
+   *
+   * In case of a rule not respected, a warning is displayed in the console.
+   */
+  checkIdsReadability(extension: AnalyzedExtension, onboarding: Onboarding) {
+    const warn = (msg: string) => {
+      console.warn(`${extension.name} Onboarding: ${msg}`);
+    };
+    onboarding.steps.forEach(step => {
+      const id = step.id;
+      const isCommand = !!step.command || !!step.component;
```
Comment (feloy) [inline packages/main/src/plugin/onboarding-registry.ts:159]: Here, I'm not sure if it is reasonable to consider a screen with a "component" to be a command.

**Review by benoitf — COMMENTED** [review]
image


I got reports when trying 👍 

about the format; maybe pickup the same than when an extension is performing a log

```
main ↪️ [redhat-sandbox] starting extension openshift-sandbox
main ↪️ [redhat-sandbox] Configfile location /Users/benoitf/.kube/config
```
so put the extension with square brackets and use the full name like `podman-desktop.compose`


Then on the log message, I think I would rework with the 'what I'm missing first' (and explanation later)

-->
`Missing 'Command' suffix for step checkComposeDownloaded that defines a command`

instead of 
`step checkComposeDownloaded defines a command, its id should end with 'Command'`

`step ${id} should end with 'View' had a different warning message template than the other


you should add unit tests so in case we tweak the isCommand, isFailedState, etc we've no-regression tests

File: packages/main/src/plugin/onboarding-registry.ts (hunk 1 above)
Comment (benoitf) [reply packages/main/src/plugin/onboarding-registry.ts:159]: hard to tell but as long as we define tests, we'll able to sort it out

**Review by benoitf — APPROVED** [review]
overall approving

**Review by deboer-tim — APPROVED** [review]
Ditto, would be good to just change the log format as Florent suggested.

File: packages/main/src/plugin/onboarding-registry.spec.ts [hunk 2]
```diff
@@ -47,131 +48,213 @@ const context = new Context(apiSender);
...
+        label: 'Installation successful',
+        title: 'Installation successful',
+        state: 'completed' as OnboardingState,
+      },
+    ],
+    enablement: 'true',
+  };
+
+  const consoleWarnMock = vi.fn();
+  console.warn = consoleWarnMock;
```
Comment (benoitf) [inline packages/main/src/plugin/onboarding-registry.spec.ts]: IMHO it should be part of a beforeEach, afterEach where in after step we reset to the previous console.warn object to not mess up further tests

**Review by benoitf — APPROVED** [review]
approving again

**Pull request merged**